### PR TITLE
Converted all .asciidoc to .md.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

While some of us do love asciidoc for its [many claimed awesome features over markdown](https://docs.asciidoctor.org/asciidoc/latest/asciidoc-vs-markdown), none of those features seem to work in "safe" modes, such as GitHub. In such, we are effectively committing documents that are hard to read or unreadable on GitHub. It's time to standardize to the most widely adopted and least common denominator, GitHub-flavored markdown.    

For the technically inclined this is what I [followed these instructions](https://alvinalexander.com/bookmarks/how-convert-asciidoc-to-html-or-markdown/):

1. Convert to .xml with `asciidoc -b docbook TESTING.asciidoc`.
2. Fix any errors in asciidoc above that `asciidoc` spews.
3. Convert xml to markdown, `pandoc -f docbook -t markdown_strict TESTING.xml -o TESTING.md`.
4. Cleanup markdown, such as 80-column layout, some syntax highlighting, add TOC, etc.
  
### Check List
--[x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1 +0,0 @@
-CHANGELOG.asciidoc  merge=union
```
Comment (dblock) [inline .gitattributes:1]: I couldn't find a CHANGELOG.asciidoc, so ...

Comment (CEHENKLE) [reply .gitattributes:1]: Huhn.  That's weird.  Vestigial I guess?

**Review by CEHENKLE — APPROVED** [review]
Thanks for doing this :)

File: .gitattributes (hunk 1 above)
Comment (adnapibar) [reply .gitattributes:1]: It was added here https://github.com/opensearch-project/OpenSearch/commit/3e7fccddaf9c20b3f2349e335bd6733916e3d2ed and then removed here https://github.com/opensearch-project/OpenSearch/commit/0f85c6429cff3f369383d256bc63c05df07bd88c

**Comment by adnapibar** [comment]
start gradle check

**Pull request merged**